### ISSUE TITLE
The compiler crashes with java.util.NoSuchElementException

### ISSUE THREAD

**Issue by janihur** [ballerina-platform/ballerina-lang#36235]
**Description:**
The compiler crashes with `java.util.NoSuchElementException`.

**Steps to reproduce:**
```
$ bal version
Ballerina 2201.0.3 (Swan Lake)
Language specification 2022R1
Update Tool 1.3.9
```
```
$ cat issue.bal 
import ballerina/http;
import ballerina/io;
import ballerina/xmldata;

public function main() returns error? {
    {
        // CASE 1

        json json1 = { 
            code: "ERROR_CODE", 
            details: "ERROR_DETAILS"
        };

        xml? xml1 = check xmldata:fromJson(json1);

        if xml1 is xml {
            io:println("xml1 is xml");
        }
        io:println(xml1);
    }
    {
        // CASE 2

        http:BadRequest badRequest = { 
            body: { 
                code: "ERROR_CODE", 
                details: "ERROR_DETAILS"
            }
        };
        io:println(badRequest);
        
        json json2 = <json> badRequest?.body;
        xml? xml2 = check xmldata:fromJson(json2);

        if xml2 is xml {
            io:println("xml2 is xml");
        }
        io:println(xml2);
    }
}
```
```
$ bal run issue.bal
Compiling source
	issue.bal
ballerina: Oh no, something really went wrong. Bad. Sad.

We appreciate it if you can report the code that broke Ballerina in
https://github.com/ballerina-platform/ballerina-lang/issues with the
log you get below and your sample code.

We thank you for helping make us better.

[2022-05-20 10:06:13,055] SEVERE {b7a.log.crash} - null 
java.util.NoSuchElementException
	at java.base/java.util.ArrayDeque.removeFirst(ArrayDeque.java:363)
	at java.base/java.util.ArrayDeque.pop(ArrayDeque.java:594)
	at io.ballerina.compiler.internal.parser.AbstractParserErrorHandler.endContext(AbstractParserErrorHandler.java:297)
	at io.ballerina.compiler.internal.parser.AbstractParser.endContext(AbstractParser.java:133)
	at io.ballerina.compiler.internal.parser.BallerinaParser.parseCompUnit(BallerinaParser.java:361)
	at io.ballerina.compiler.internal.parser.BallerinaParser.parse(BallerinaParser.java:100)
	at io.ballerina.compiler.syntax.tree.SyntaxTree.from(SyntaxTree.java:55)
	at io.ballerina.projects.DocumentContext.parse(DocumentContext.java:86)
	at io.ballerina.projects.DocumentContext.syntaxTree(DocumentContext.java:90)
	at io.ballerina.projects.DocumentContext.getModuleLoadRequests(DocumentContext.java:127)
	at io.ballerina.projects.DocumentContext.moduleLoadRequests(DocumentContext.java:121)
	at io.ballerina.projects.ModuleContext.populateModuleLoadRequests(ModuleContext.java:220)
	at io.ballerina.projects.PackageResolution.getModuleLoadRequestsOfDirectDependencies(PackageResolution.java:222)
	at io.ballerina.projects.PackageResolution.resolveSourceDependencies(PackageResolution.java:266)
	at io.ballerina.projects.PackageResolution.buildDependencyGraph(PackageResolution.java:214)
	at io.ballerina.projects.PackageResolution.<init>(PackageResolution.java:94)
	at io.ballerina.projects.PackageResolution.from(PackageResolution.java:101)
	at io.ballerina.projects.PackageContext.getResolution(PackageContext.java:242)
	at io.ballerina.projects.Package.getResolution(Package.java:149)
	at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:80)
	at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
	at io.ballerina.cli.cmd.RunCommand.execute(RunCommand.java:203)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.ballerina.cli.launcher.Main.main(Main.java:51)
```

**Comment by lochana-chathura** [comment]
Causing due to an issue parsing the following syntax. 
[1]
```bal
function main() {
    {
        http:BadRequest badRequest = {};
    }
}
```

Workaround is to use `do-stmt` instead of `block-stmt` as they both the same. 
[2]
```bal
function main() {
    do {
        http:BadRequest badRequest = {};
    }
}
```

[1] is going to be disallowed anyway with #36187.

**Comment by lochana-chathura** [comment]
Fixed in #36297
